selftests/functional/test_basic.py

```diff
@@ -105,7 +107,11 @@ def probe_binary(binary):
 
 
 CC_BINARY = probe_binary('cc')
+# On macOS, the default coreutils installation (brew)
+# installs the gnu utility versions with a g prefix.
 ECHO_BINARY = probe_binary('echo')
+if ECHO_BINARY is not None:
+    ECHO_BINARY = probe_binary('gecho')
```

Comment:
The problem is that the BSD echo doesn't have the -e flag. Looking at this again, I notice that this will not work the way I expected, since normally the BSD echo is used (it exists) instead of the GNU one.